Packages/com.unity.inputsystem/InputSystem/Actions/InputActionSet.cs

```diff
@@ -185,6 +185,17 @@ public InputActionSet Clone()
                 m_Actions = ArrayHelpers.Clone(m_Actions)
             };
 
+            if(clone.m_Actions != null)
+            {
+                // ensure cloned actions refer to the new set and are not singletons
+                var cloneActions = clone.m_Actions;
+                var length = cloneActions.Length;
+                for(int i = 0; i < length; ++i)
+                {
+                    cloneActions[i].m_ActionSet = clone;
+                }
+            }
```

Comment:
Took me a moment to realize but this unfortunately is a symptom fix. `ArrayHelpers.Clone` is calling `Clone` on each action which results in the action set being populated with actions that think they're singleton actions instead of being part of a set. I.e. the root problem is that the set isn't cloning actions correctly.

Good catch, though. The action stuff is currently undergoing a bigger refactor and I'll make sure to address this problem as part of that.